src/mostlyai/qa/sampling.py

```diff
@@ -67,22 +67,24 @@ def pull_data_for_accuracy(
     assert setup is None or setup in ["1:1", "1:N"]
 
     key = "__KEY"
+    max_sample_size = 10_000_000 if max_sample_size is None else max_sample_size
```

Comment:
I'm not entirely sure, why do we need this cap of `10_000_000`, but is that you don't want to pass `None` to `head`? `.head(None)` is a valid call, and it means taking everything.